purchases/src/main/kotlin/com/revenuecat/purchases/caching/DeviceCache.kt

```diff
@@ -3,28 +3,82 @@
 //  Copyright © 2019 RevenueCat, Inc. All rights reserved.
 //
 
-package com.revenuecat.purchases
+package com.revenuecat.purchases.caching
 
 import android.content.SharedPreferences
+import com.revenuecat.purchases.Offerings
+import com.revenuecat.purchases.PurchaseWrapper
+import com.revenuecat.purchases.PurchaserInfo
+import com.revenuecat.purchases.Purchases
+import com.revenuecat.purchases.buildPurchaserInfo
+import com.revenuecat.purchases.debugLog
+import com.revenuecat.purchases.sha1
 
 import org.json.JSONException
 import org.json.JSONObject
 import java.util.Date
 
 private const val CACHE_REFRESH_PERIOD = 60000 * 5
 
-internal class DeviceCache(
-    private val preferences: SharedPreferences,
-    apiKey: String
-) {
-    val legacyAppUserIDCacheKey = "com.revenuecat.purchases.$apiKey"
-    val appUserIDCacheKey = "com.revenuecat.purchases.$apiKey.new"
+internal class DeviceCache {
+    private val preferences: SharedPreferences
+    private val offeringsCachedObject: InMemoryCachedObject<Offerings>
+
+    val legacyAppUserIDCacheKey: String
+    val appUserIDCacheKey: String
     private val attributionCacheKey = "com.revenuecat.purchases.attribution"
-    val tokensCacheKey = "com.revenuecat.purchases.$apiKey.tokens"
+    val tokensCacheKey: String
+
+    var purchaserInfoCachesLastUpdated: Date? = null
+
+    constructor(
+        preferences: SharedPreferences,
+        apiKey: String,
+        offeringsCachedObject: InMemoryCachedObject<Offerings>
+    ) {
+        this.preferences = preferences
+        this.offeringsCachedObject = offeringsCachedObject
+        this.legacyAppUserIDCacheKey = "com.revenuecat.purchases.$apiKey"
```

Comment:
why have these keys separate from the `attributionCacheKey`? Also, could we extract the common prefix?